graph.cpp

```diff
@@ -20,11 +20,15 @@ Graph::Graph(uint64_t num_nodes): num_nodes(num_nodes) {
     parent[i] = i;
   }
   num_updates = 0; // REMOVE this later
-
   std::string buffer_loc_prefix = configure_system(); // read the configuration file to configure the system
+#ifdef USE_FBT_F
   // Create buffer tree and start the graphWorkers
   bf = new BufferTree(buffer_loc_prefix, (1<<20), 8, num_nodes, GraphWorker::get_num_groups(), true);
   GraphWorker::start_workers(this, bf);
+#else
+  wq = new WorkQueue(1 << 10, num_nodes, 2*GraphWorker::get_num_groups());
```

Comment:
Is 1 << 10 the size of the buffer?
If so, that's too small. The size of each node sketch is much larger than a kilobyte once the number of nodes is greater than 11. We want the buffer to be the size of a node sketch, or 24 * log_2^3(N) rounded to the nearest Node.
Instead of a static approach it's probably worth while to calculate the value we want.